biz.aQute.resolve/test/biz/aQute/resolve/ValidatorTest.java

```diff
@@ -64,4 +69,112 @@ public void _testLarger() throws Exception {
 			assertTrue(validator.check());
 		}
 	}
+
+	public void testIndexAgainstLargeTargetPlatform() throws Exception {
+		try (ResolverValidator validator = new ResolverValidator();) {
+			ResourceBuilder resourceBuilder = new ResourceBuilder();
+
+			resourceBuilder.addEE(EE.JavaSE_1_7);
+			resourceBuilder.addManifest(OSGI_CORE.R6_0_0.getManifest());
+
+			// Create a rich base platform by adding more resources.
+
+			File[] files = new File[] {
+					IO.getFile("testdata/liferay-indexes/collaboration-1.0-index.xml"),
+					IO.getFile("testdata/liferay-indexes/forms-and-workflow-1.0-index.xml"),
+					IO.getFile("testdata/liferay-indexes/foundation-1.0-index.xml"),
+					IO.getFile("testdata/liferay-indexes/target.platform.index.xml"),
+			};
+
+			for (File file : files) {
+				try (XMLResourceParser xmlResourceParser = new XMLResourceParser(file.toURI())) {
```

Comment:
This loop looks wrong. You are creating a single Resource from the capabilities and requirements of all the of the resources in all of the XML files. That seems like it would create a possible bad resource that could have conflicting or invalid state.

If your goal is to create a system bundle, which seems to be what you are doing since you are starting with an EE and the Core API, then how can this system bundle import packages or require random capabilities which are sure to happen since to add all requirements to the resource?